apps/ui/components/application/TimeSelector.tsx

```diff
@@ -370,7 +408,16 @@ const TimeSelector = ({
   return (
     <>
       <OptionWrapper>
-        <PrioritySelect
+        <StyledSelect
+          id={`time-selector__select--reservation-unit-${index}`}
+          aria-labelledby={t("appplication:Page2.")}
+          options={reservationUnitOptions}
+          defaultValue={reservationUnitOptions[selectedReservationUnit]}
```

Comment:
Is there a reason to set `defaultValue` and not `value`?
Technically it shouldn't matter right now, but if it's `defaultValue` I'd rename the `selectedReservationUnit` so it's clear that it can't be controlled after initialisation.

Another issue with `defaultValue` is that Queries are async, so unless the component blocks till we have query data the `defaultValue` will always be undefined.

Reply:
No real reason. It takes after the priority selector which was already there - `value` is probably better.